File: Py/189-rotate-array/solution1.py
class Solution:
    def rotate(self, nums: List[int], k: int) -> None:
        """
        Rotate the list 'nums' to the right by 'k' steps.

        Args:
        - nums (List[int]): List of integers to be rotated.
        - k (int): Number of steps to rotate the list.

        Returns:
        None. The list 'nums' is modified in-place.
        """
        n = len(nums)
        ans = [0] * n  # Initialize a list of zeros with size 'n'

        for i in range(n):
            ans[(i + k) % n] = nums[i]

        nums[:] = ans  # Update the original 'nums' list with rotated values

File: Cpp/189-rotate-array/solution1.cpp
class Solution {
public:
    // TC: O(n)
    // SC: O(n)
    void rotate(vector<int>& nums, int k) {
        // Get the size of the 'nums' vector
        int n = nums.size();

        // Create a new vector 'ans' to store the rotated values
        vector<int> ans(n);

        // Iterate over the 'nums' vector
        for(int i=0; i<n; i++){
            // For each number in 'nums', find its new position after rotation
            // The new position is determined by '(i+k)%n'
            // This ensures the rotation wraps around if 'k' is larger than 'n'
            ans[(i+k)%n] = nums[i];
        }

        // Update the original 'nums' vector with the rotated values from 'ans'
        nums = ans;
    }
};
